### PR TITLE
React translation enablement

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Implement the translation functionality of the react `t()` function using `node-gettext`

## GUI diff

After:

Language picker added for user preferences and user creation interface.
language_picker

- [x] **DONE**

## Documentation

- Documentation issue was created: https://github.com/SUSE/spacewalk/issues/12301

- [x] **DONE**

## Test coverage
- No tests: 

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/11667

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java [hunk 1]
```diff
@@ -329,6 +330,11 @@
      */
     public static final String ALLOW_ADDING_PATCHES_VIA_API = "java.allow_adding_patches_via_api";
 
+    /**
+     * Specify the default language to use if user preferences are not available such as on the login page
+     */
+    public static final String PREFERRED_LOCALE = "web.locale";
```
Comment (ncounter) [inline java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java:336]: why not `DEFAULT_LOCALE`? btw, is this going to affect the language or also all the localization like date and time?

File: java/code/webapp/WEB-INF/decorators/layout_head.jsp [hunk 2]
```diff
@@ -62,14 +62,17 @@
         </c:otherwise>
     </c:choose>
 
+    <!-- expose user preferred language to the application -->
+    <c:set var="currentLocale" value="${GlobalInstanceHolder.MENU_TREE.getCurrentLocale(pageContext)}"/>
+    <script> preferredLocale='${currentLocale}'</script>
```
Comment (ncounter) [inline java/code/webapp/WEB-INF/decorators/layout_head.jsp:67]: Can we put this into `window.preferredLocale`?

File: java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java (hunk 1 above)
Comment (parlt91) [reply java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java:336]: I guess you are right `DEFAULT_LOCALE` is better in this context.
It is just the language.

File: java/code/webapp/WEB-INF/decorators/layout_head.jsp (hunk 2 above)
Comment (parlt91) [reply java/code/webapp/WEB-INF/decorators/layout_head.jsp:67]: Yes of course we can. I will change it

File: branding/css/susemanager-theme.less [hunk 3]
```diff
@@ -896,10 +896,6 @@ input[name="cveIdentifierId"]#cveIdentifierId {
 .popover-content ul li {
   margin-top: .5em
 }
-/* Create New User. Language selection hidden for SUSE Manager users as we only offer support for English */
```
[the remark this answers is not in the document]
Comment (parlt91) [reply branding/css/susemanager-theme.less:899]: Yeah I was shocked when I saw this :sweat_smile:

**Comment by parlt91** [comment]
@cbbayburt Can you also have a look at the PR?

**Review by cbbayburt — APPROVED** [review]
Looks all good to me :+1: 
Thanks for having well-defined commits!

Just a trivial question below, feel free to ignore :)

File: java/code/src/com/suse/manager/webui/controllers/login/templates/login.jade [hunk 4]
```diff
@@ -58,6 +58,7 @@ html
             | #{request_method}
 
         script(type='text/javascript').
+            preferredLocale = "#{preferredLocale}";
```
[the remark this answers is not in the document]
Comment (parlt91) [reply java/code/src/com/suse/manager/webui/controllers/login/templates/login.jade:60]: It seems that it is already added to the window variable. At least typing `window.preferredLocale` into the console at the login page returns the value

**Review by ncounter — APPROVED** [review]

**Pull request merged**